### PR TITLE
Added setup documentation

### PR DESCRIPTION
**Pull request by pwilkins**
#### What are the relevant tickets?
Issue #77 -  Clean instructions in README for docker-osx-dev

#### What's this PR do?
Provide guidance for developers trying run Micromasters.

#### Where should the reviewer start?
Open the "Getting Started" section in ``README.md`` and follow the instructions.

#### How should this be manually tested?
Follow the instructions and see if it works.

#### Any background context you want to provide?
The setup procedure is more complex for OSX than for Linux.

#### What GIF best describes this PR or how it makes you feel?

Added setup documentation to ``README.md``.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,104 @@
...
+Due to issues using Docker in OSX, development is more efficient if you
+install `docker-osx-dev <https://github.com/brikis98/docker-osx-dev>`_.
+``docker-osx-dev`` keeps host file system changes synced to the Docker
+container. You can install it by typing ``make`` once you have created the
+Docker container.
+
+Subsequently, before you start your Docker container with
+``docker-compose up``, you would run::
+  
+     docker-osx-dev -m default -s ./
```
Comment (noisecapella) [inline README.md]: I think we'll want to change `default` to `mm` here and in the Makefile, but we can do this in a separate PR.

Can you add ` --ignore-file '.rsync-ignore'` to the end of this line?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,104 @@
...
+#. (OSX only) Create your docker machine::
+
+     docker-machine create default
+     docker-machine start default
+     docker-machine env default
+     eval "$(docker-machine env default)"
+
+   These commands create a Docker container named ``default``, start the
+   container, and configure environment variables to facilitate communication
+   with ``ccxcon``.
```
Comment (noisecapella) [inline README.md]: This should refer to the edX instance, not CCXCon

File: README.md [hunk 3]
```diff
@@ -1,2 +1,104 @@
...
+
+#. Create a user who has permission to view the course::
+
+     docker-compose run web python manage.py createsuperuser
+
+#. Start the machine::
+
+     docker-compose up
+
+#. visit: http://192.168.99.100:8075/
```
Comment (noisecapella) [inline README.md]: Can you add `(OSX only)` here and remove it below? Also, change the port number from `8075` to `8079`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,104 @@
...
+
+  Your IP address may vary depending on what address docker assigns to your VM.
+  If the IP above doesn't work:
+
+  1. Power down your docker-machine (``docker-machine stop default``)
+  2. ``docker-machine ssh default``
+  3. Once in the docker machine shell, run ``ifconfig``, and look for
+     the IP address that begins with ``192.168.99.xxx``.
+
+- Click the login button on the TP UI and put in the user credentials
```
Comment (noisecapella) [inline README.md]: Can you remove the lines here and below? It looks like it only applies to teachersportal

File: README.md [hunk 5]
```diff
@@ -1,2 +1,104 @@
...
+
+#. Start the machine::
+
+     docker-compose up
+
+#. visit: http://192.168.99.100:8075/
+
+.. note:: (OSX only)
+
+  Your IP address may vary depending on what address docker assigns to your VM.
```
Comment (noisecapella) [inline README.md]: Technically it's the VM that's requesting the IP address, not docker. Can you change this to `depending on which IP address your VM uses`?

**Comment by pdpinch** [comment]
@pwilkins is this ready for review?

**Comment by pdpinch** [comment]
Why is this labelled "waiting on author" ?

**Comment by pwilkins** [comment]
Changed label to "Needs Review".

File: README.md [hunk 6]
```diff
@@ -1,2 +1,171 @@
...
+
+(OSX only) visit: http://192.168.99.100:8079/
+
+  Note: Your IP address may vary depending on what address docker 
+  assigns to your VM.  If the IP address above doesn't work:
+
+  1. Power down your docker-machine (``docker-machine stop default``)
+  2. ``docker-machine ssh default``
+  3. Once in the docker machine shell, run ``ifconfig``, and look for
+     the IP address that begins with ``192.168.99.xxx``.
```
Comment (giocalitri) [inline README.md]: there is a quicker way to get the IP that is
`docker-machine ls`

this will give an output like
```
NAME             ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER    ERRORS
ccxcon           -        virtualbox   Stopped                                       Unknown
dev              -        virtualbox   Stopped                                       Unknown
lore             -        virtualbox   Stopped                                       Unknown
micromasters     -        virtualbox   Running   tcp://192.168.99.100:2376           v1.10.2
teachersportal   -        virtualbox   Stopped                                       Unknown
```

**Comment by giocalitri** [comment]
a very minor comment but :+1:

File: README.md (hunk 6 above)
Comment (pwilkins) [inline README.md]: Nice!
